slamd/discovery/processing/algorithms/plot_generator.py

```diff
@@ -1,35 +1,76 @@
 import json
+import numpy as np
 import pandas as pd
 import plotly
 import plotly.express as px
+import plotly.graph_objects as go
+from plotly.subplots import make_subplots
 from sklearn.manifold import TSNE
 
+UNCERTAINTY_COLUMN_PREFIX = 'Uncertainty ('
+
 
 class PlotGenerator:
 
     @classmethod
     def create_target_scatter_plot(cls, plot_df):
-        dimensions = [col for col in plot_df.columns if col != 'Utility' and col != 'Row number']
+        dimensions = list(filter(lambda col: col != 'Utility' and col !=
+                          'Row number' and not col.startswith(UNCERTAINTY_COLUMN_PREFIX), plot_df.columns))
```

Comment:
Why can `and not col.startswith(UNCERTAINTY_COLUMN_PREFIX)` not be used in the list comprehension? Would be easier to understand than a lambda. Since you have three conditions, consider turning this into an actual loop. Also, consider only filtering for the column prefix, and then dropping Utility and Row Number by name, since there is only one of each.